### ISSUE TITLE
WithDefaultServiceConfig accepts only raw JSON

### ISSUE THREAD

**Issue by kazegusuri** [grpc/grpc-go#3003]
### Use case(s) - what problem will this feature solve?

It seems WithBalancerName is deprecated in https://github.com/grpc/grpc-go/pull/2900 in favor of WithDefaultServiceConfig. WithDefaultServiceConfig accepts only raw JSON, so if we want to use specific balancer statically, we need to use JSON to specify that.

### Proposed Solution

Please keep WithBalancerName as is, or another way to specify the configurations instead of using JSON.

**Comment by kazegusuri** [comment]
@dfawley How do you think of this? Not everyone use ServiceConfig with discovery, so it would be easier if we have another way to specify configs without specifying raw json.

**Comment by menghanl** [comment]
The main reason to deprecate `WithBalancerName` is to avoid having two ways to do the same thing.
Reason for `json` is that it's the service config format agreed upon by all the grpc languages.

It is reasonable to support specifying service using a format other than json, maybe a native go struct.

For now, please try `WithDefaultServiceConfig`. To set balancer to roundrobin, you can set 
```
grpc.WithDefaultServiceConfig(`{"loadBalancingPolicy":"round_robin"}`)
```

**Comment by kazegusuri** [comment]
What's the status on this?

**Comment by zcong1993** [comment]
any update?

**Comment by dfawley** [comment]
This is unfortunately not something the team has resources to spend on right now.  This will require us to evaluate the existing [`grpc.ServiceConfig`](https://pkg.go.dev/google.golang.org/grpc#ServiceConfig) to determine if we should just keep using it, or finish removing it and replace it with something more appropriate.

One reason this is not a priority is that the JSON form should be completely functional to use. 
Note that there is a [proto representation for service config](https://github.com/grpc/grpc-proto/blob/master/grpc/service_config/service_config.proto), and it's possible to serialize that to JSON and inject it, so there already technically is a native Go struct for service config.

**Comment by seeruk** [comment]
> Perhaps we should link to it from our godoc comments on the WithDefaultServiceConfig function, and then it would be more discoverable and a testament to its freshness, while also improving our documentation. What do you think?

Yeah, that's a good shout. Removing that experimental comment would also help too, I agree.

> FWIW we do have an example of setting exactly this here:

If I'm understanding correctly, that actually is a deprecated approach. The one I landed on was this:

```go
grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin":{}}]}`)
```

**Comment by dfawley** [comment]
> If I'm understanding correctly, that actually is a deprecated approach

Argh, I didn't even look closely at it, but you are right.  What you have is the preferred form, and we should update our example.  Note that, in this case, "deprecated" means "there's another way, which we recommend using instead", not "it will be removed", since we intend to keep backward compatibility forever (for anything not originally marked w/experimental).
